Tests/AfluentTests/SequenceTests/DeferredSequenceTests.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  DeferredSequenceTests.swift
+//
+//
+//  Created by Annalise Mariottini on 12/20/23.
+//
+
+import Afluent
+import Foundation
+import XCTest
+
+class DeferredSequenceTests: XCTestCase {
+    func testUpstreamSequenceDefersExecutionUntilIteration() async throws {
+        let shouldNotStartExpectation = expectation(description: "sequence not started")
+        shouldNotStartExpectation.isInverted = true
+        let shouldStartExpectation = expectation(description: "sequence started")
+
+        let sent = Array(0...9)
+
+        let sequence = DeferredAsyncSequence {
+            defer {
+                shouldNotStartExpectation.fulfill()
+                shouldStartExpectation.fulfill()
+            }
+            return AsyncStream(Int.self, { continuation in
+                sent.forEach { continuation.yield($0) }
+                continuation.finish()
+            })
+        }
+
+        await fulfillment(of: [shouldNotStartExpectation], timeout: 0.01)
+
+        var iterator = sequence.makeAsyncIterator()
+
+        await fulfillment(of: [shouldStartExpectation], timeout: 0)
+
+        var received: [Int] = []
```

Comment:
You could probably use collect()

Reply:
Nope, sorry about that. I built `collect` in a research branch forever ago. Do not feel obligated to build it for this MR.